Tools/autotest/arducopter.py

```diff
@@ -7769,39 +7945,257 @@ def AP_Avoidance(self):
         self.context_pop()
 
     def PAUSE_CONTINUE(self):
+
+        # load auto mission
         self.load_mission("copter_mission.txt", strict=False)
 
+        # start mission
         self.set_parameter("AUTO_OPTIONS", 3)
-        self.change_mode('AUTO')
+        self.change_mode("AUTO")
         self.wait_ready_to_arm()
         self.arm_vehicle()
 
-        self.wait_waypoint(4, 4)
-        self.run_cmd(mavutil.mavlink.MAV_CMD_DO_PAUSE_CONTINUE,
-                     0, # param1
-                     0, # param2
-                     0, # param3
-                     0, # param4
-                     0, # param5
-                     0, # param6
-                     0 # param7
-                     )
+        # wait for the waypoint
+        self.wait_current_waypoint(3, 500)
+
+        # send pause command
+        self.cmd_pause()
 
+        # wait vehicle to pause
         self.wait_groundspeed(0, 1, minimum_duration=5)
 
-        self.run_cmd(mavutil.mavlink.MAV_CMD_DO_PAUSE_CONTINUE,
-                     1, # param1
-                     0, # param2
-                     0, # param3
-                     0, # param4
-                     0, # param5
-                     0, # param6
-                     0 # param7
-                     )
+        # send resume command
+        self.cmd_resume()
 
-        self.wait_groundspeed(5, 100)
+        # wait for the waypoint
+        self.wait_current_waypoint(4, 500)
 
-        self.wait_disarmed()
+        # send pause command
```

Comment:
Let's keep this test simple, just attempt to pause along a waypoint segment and ensure it pauses.  The resume and make sure it restarts.  Of course also make sure that if the vehicle doesn't actually pause or resume that the test fails.  We shouldn't need to pause and resume twice though.

Reply:
Mmmm... doing it twice covers the "0, 1 or many" adage, better, 'though.

It can catch a lack of reinitialisation, for example.